challenge-sizebay/src/Contexts/GlobalContext.jsx

```diff
@@ -0,0 +1,3 @@
+import { createContext } from "react";
+
+export const GlobalContext = createContext()
```

Comment:
nit: you could also create a custom hook to wrap your context call.

Like `useApp()` that inherits `useContext(GlobalContext)`. This would increase DX and avoid verbosity over a one-liner.

Reply:
I don't know if I understood exactly what you mean.
- Isn't it still going to be necessary to have a separate file where I create the context? 
- And also, as I was testing your idea, it was still necessary to receive the GlobalContext as a prop in this new useApp hook I created. I thought the goal was to avoid importing the GlobalContext everywhere, but it was still necessary. 
- It seems that the hook is not being reached by the provider, so the data I'm trying to return is `undefined` and I can't de-structure it to return the stuff I need.

I left the hook call in `addNewItem.jsx`
Maybe I wrote it wrong? 🤔 
I'm gonna push the code for the custom hook I created so you can take a look

If you reply something today, I'll be able to work on the changes you tell me :)

Reply:
@luizfbarbosa12 the main intention here is to avoid importing every Context related module from React on the components that'll inherit it. 

Instead, you can create a custom hook called `useApp` that wraps the context calls and simplifies the module import.

The issue that turned your context to return `undefined` is that you're declaring `createContext` with empty arguments (no initial state). It's fine given the scenario, and shouldn't affect your refactor on creating a custom context hook.

Try doing something similar to this code below:

```js
export const useAppContext = () => {
  const context = useContext(AppContext)

  return context
}

// SomeComponent.tsx

import { useAppContext } from 'your/path/file.ts'

const app = useAppContext()

// do your magic here =)
```

Lmk what you think.

Reply:
I spent some time trying to understand, but it's still very confusing :/ 
I have no idea where to do each of the declarations and I can't find anything like that online. 

I think I'll have to let this one pass this time 😞

Reply:
It's no worries; I can help you with that. Leave this refactor open and let's talk next week about how this can be improved. Sounds good?

Reply:
Awesome!